infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -119,6 +124,12 @@ public LogReplicationMetadataManager(CorfuRuntime runtime, long topologyConfigId
         }
     }
 
+    @VisibleForTesting
+    public LogReplicationMetadataManager(CorfuRuntime runtime, long topologyConfigId, boolean isLeader) {
+        this(runtime, topologyConfigId);
+        setLeadership(isLeader);
+    }
+
```

Comment:
its an atomicBoolean, so this way is better.
If we have a setter, we'll need to create atomicBoolean objects and then pass that object to the setter

Reply:
thanks I see the point, then I thought about 2 questions here:
* For constructor - previously I was thinking we could avoid creating this testing-purpose constructor and just do normal `new LogReplicationMetadataManager` and then `metadataManager.setLeaderShip()`, since the set method is already public.  which is just a minor thing out of my personal preference
* For that `AtomicBoolean` field - I agree and also we should avoid that Getter annotation as well right? since it will expose the AtomicBoolean's reference and the caller could change its state then. should we wrap it up like `setLeadership` and return primitive boolean?

Reply:
I see your point for (1). I am changing it to remove the new constructor like you suggest
as for (2), I left the "Getter", but made it a "final" variable. Getter is safe now.